index.js

```diff
@@ -22,47 +24,47 @@ var generateHTML = function (swaggerDoc, opts, options, customCss, customfavIcon
-    customeSiteTitle = customeSiteTitle || 'Swagger UI';
-	var html = fs.readFileSync(__dirname + '/indexTemplate.html.tpl');
-    try {
-    	fs.unlinkSync(__dirname + '/index.html');
-    } catch (e) {
+  options = options || {}
+  var explorerString = isExplorer ? '' : '.swagger-ui .topbar .download-url-wrapper { display: none }'
+  customCss = explorerString + ' ' + customCss || explorerString
+  customfavIcon = customfavIcon || false
+  customeSiteTitle = customeSiteTitle || 'Swagger UI'
+  fs.unlink(__dirname + '/index.html', function(err) {
+    console.error(err)
+  })
+
+  var favIconString = customfavIcon ? '<link rel="icon" href="' + customfavIcon + '" />' : favIconHtml
+  var htmlWithCustomCss = htmlTplString.toString().replace('<% customCss %>', customCss)
+  var htmlWithFavIcon = htmlWithCustomCss.replace('<% favIconString %>', favIconString)
+  var htmlWithCustomJs = htmlWithFavIcon.replace('<% customJs %>', customJs ? `<script src="${customJs}"></script>` : '')
+  var htmlWithCustomCssUrl = htmlWithCustomJs.replace('<% customCssUrl %>', customCssUrl ? `<link href="${customCssUrl}" rel="stylesheet">` : '')
 
-    }
-
-    var favIconString = customfavIcon ? '<link rel="icon" href="' + customfavIcon + '" />' : favIconHtml;
-    var htmlWithCustomCss = html.toString().replace('<% customCss %>', customCss);
-    var htmlWithFavIcon = htmlWithCustomCss.replace('<% favIconString %>', favIconString);
-    var htmlWithCustomJs = htmlWithFavIcon.replace('<% customJs %>', customJs ? `<script src="${customJs}"></script>` : '');
-    var htmlWithCustomCssUrl = htmlWithCustomJs.replace('<% customCssUrl %>', customCssUrl ? `<link href="${customCssUrl}" rel="stylesheet">` : '');
+  var initOptions = {
+    swaggerDoc: swaggerDoc || undefined,
+    customOptions: options,
+    swaggerUrl: swaggerUrl || undefined,
+    swaggerUrls: swaggerUrls || undefined
+  }
 
-    var initOptions = {
-      swaggerDoc: swaggerDoc || undefined,
-      customOptions: options,
-      swaggerUrl: swaggerUrl || undefined,
-      swaggerUrls: swaggerUrls || undefined
-    }
-    var js = fs.readFileSync(__dirname + '/swagger-ui-init.js.tpl');
-    swaggerInit = js.toString().replace('<% swaggerOptions %>', stringify(initOptions))
-    return htmlWithCustomCssUrl.replace('<% title %>', customeSiteTitle)
+  swaggerInit = jsTplString.toString().replace('<% swaggerOptions %>', stringify(initOptions))
+  return htmlWithCustomCssUrl.replace('<% title %>', customeSiteTitle)
 }
 
 var setup = function (swaggerDoc, opts, options, customCss, customfavIcon, swaggerUrl, customeSiteTitle) {
-    var htmlWithOptions = generateHTML(swaggerDoc, opts, options, customCss, customfavIcon, swaggerUrl, customeSiteTitle)
-    return function (req, res) { res.send(htmlWithOptions) };
-};
+  return function (req, res) {
+    res.send(
+      generateHTML(swaggerDoc || req.swaggerDoc, opts, options, customCss, customfavIcon, swaggerUrl, customeSiteTitle, htmlTplString, jsTplString)
+    )
```

Comment:
can the generateHTML function be called outside the  ```return function (req, res) { ```. It just seems a little overkill to have to regenerate the HTML again with each request. I think having the if statement you had in the previous commit is a nice compromise